src/test/java/org/broadinstitute/hellbender/tools/picard/sam/GatherBamFilesIntegrationTest.java

```diff
@@ -40,6 +57,21 @@ public void testTheGathering() throws Exception {
         args.add(outputFile.getAbsolutePath());
         runCommandLine(args);
         SamAssertionUtils.assertSamsEqual(ORIG_BAM, outputFile);
-        SamAssertionUtils.assertSamsNonEqual(ORIG_BAM, SPLIT_BAMS.get(0)); // sanity check
+        Assert.assertNotNull(SamAssertionUtils.samsEqualStringent(ORIG_BAM, SPLIT_BAMS_UNMAPPED_LAST.get(0), ValidationStringency.DEFAULT_STRINGENCY, null)); // sanity check
     }
+
+    @Test
+    public void testTheGatheringUnmappedFirst() throws Exception {
```

Comment:
Use a `DataProvider` to provide the `SPLIT_BAMS_UNMAPPED_FIRST` and `SPLIT_BAMS_UNMAPPED_LAST` lists, and combine this test with the previous test.